src/django/api/views/v1/moderation_events.py

```diff
@@ -49,3 +52,34 @@ def list(self, request):
             query_body
         )
         return Response(response)
+
+    @handle_errors_decorator
+    def retrieve(self, request,  pk=None):
+        if not IsRegisteredAndConfirmed().has_permission(request, self):
```

Comment:
Can you use `permission_classes = [IsRegisteredAndConfirmed]` here as it works for `list()` method above?